### PR TITLE
Origin controller

### PR DESCRIPTION
**Pull request by alcalyn**
```
1. composer install
2. Update parameters.yml according to parameters.yml.dist
3. php bin/console doctrine:database:create --env=test
4. php bin/console doctrine:schema:update --force --env=test
5. php vendor/bin/phpunit 
```

Urls to test:
```
POST   /origins    {"name":"sncf"} //application/json
GET    /origins
GET    /origins/1
PATCH  /origins/1    {"name":"sncf_edited"} //application/json
DELETE /origins/1
```

### CODE REVIEW

File: src/CoreBundle/Controller/OriginController.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+     *
+     * @return View
+     */
+    public function getOriginAction(Origin $origin)
+    {
+        return View::create($origin);
+    }
+
+    /**
+     * @Rest\RequestParam(name="name", requirements="[a-z_\.]+")
```
Comment (Netmisa) [inline src/CoreBundle/Controller/OriginController.php]: We need this field to create origin. You forget "strict=true" option.

File: src/CoreBundle/Controller/OriginController.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+        $em = $this->getDoctrine()->getManager();
+
+        try {
+            $em->flush();
+        } catch (UniqueConstraintViolationException $e) {
+            throw new ConflictHttpException($e->getMessage(), $e);
+        }
+
+        return View::create(null, Response::HTTP_NO_CONTENT);
```
Comment (Netmisa) [inline src/CoreBundle/Controller/OriginController.php]: Why you doesn't return object updated ?

File: src/CoreBundle/Controller/OriginController.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    /**
+     * @param Origin $origin
+     *
+     * @return View
+     */
+    public function deleteOriginsAction(Origin $origin)
+    {
+        $em = $this->getDoctrine()->getManager();
+
+        $em->remove($origin);
```
Comment (Netmisa) [inline src/CoreBundle/Controller/OriginController.php]: This is work if $origin equal null ?
I thinks it's better to add exception with NotFoundHttpException.

File: src/CoreBundle/Controller/OriginController.php (hunk 1 above)
Comment (alcalyn) [inline src/CoreBundle/Controller/OriginController.php]: raise 404 before controller: http://symfony.com/doc/current/bundles/SensioFrameworkExtraBundle/annotations/converters.html

File: src/CoreBundle/Controller/OriginController.php (hunk 2 above)
Comment (alcalyn) [inline src/CoreBundle/Controller/OriginController.php]: No needed

File: src/CoreBundle/Controller/OriginController.php (hunk 1 above)
Comment (alcalyn) [inline src/CoreBundle/Controller/OriginController.php]: Not needed, already raise 400 by paramfetecher

File: src/CoreBundle/Controller/OriginController.php (hunk 3 above)
Comment (alcalyn) [inline src/CoreBundle/Controller/OriginController.php]: raise 404 before controller: http://symfony.com/doc/current/bundles/SensioFrameworkExtraBundle/annotations/converters.html

Comment (Netmisa) [inline src/CoreBundle/Controller/OriginController.php]: Ok tanks

File: src/CoreBundle/Controller/OriginController.php (hunk 1 above)
Comment (Netmisa) [inline src/CoreBundle/Controller/OriginController.php]: Ok, and we doesn't want to be nullable.

**Pull request merged**